tests/invalid_models_tests/test_models.py

```diff
@@ -618,6 +618,36 @@ class Meta:
 
         self.assertEqual(Bar.check(), [])
 
+    def test_expression_unique_constraint(self):
+        class Model(models.Model):
+            name = models.CharField(max_length=10)
+
+            class Meta:
+                constraints = [models.UniqueConstraint(Lower('name'), name='lower_name_unq')]
+
+        warn = Warning(
+            '%s does not support unique constraints on expressions.'
+            % connection.display_name,
+            hint=(
+                "A constraint won't be created. Silence this warning if you don't "
+                "care about it."
+            ),
+            obj=Model,
+            id='models.W044',
+        )
+        expected = [] if connection.features.supports_expression_indexes else [warn]
+        self.assertEqual(Model.check(databases=self.databases), expected)
+
+    def test_expression_unique_constraint_required_db_features(self):
+        class Model(models.Model):
+            name = models.CharField(max_length=10)
+
+            class Meta:
+                constraints = [models.UniqueConstraint(Lower('name'), name='lower_name_unq')]
+                required_db_features = {'supports_expression_indexes'}
+
+        self.assertEqual(Model.check(databases=self.databases), [])
```

Comment:
Doesn't this assertion fail if the database doesn't support expression indexes?

Reply:
This is actually a check that the `required_db_features` attribute works as intended for databases not supporting this feature: https://github.com/django/django/blob/4d8c9e83550a394af8f3c6ecc4b576c0e4e4152c/django/db/models/base.py#L2042-L2047

Reply:
Oh! I see! Maybe worth a comment?